File: Python/482. License Key Formatting.py
class Solution:
    def licenseKeyFormatting(self, S, K):
        """
        :type S: str
        :type K: int
        :rtype: str
        """
        s = S.upper().replace('-', '')
        last, ans = K, str()
        if len(s) % K != 0:
            last = len(s) % K
        ans = s[:last]
        for i in range(last, len(s), K):
            ans += '-' + s[i:i+K]
        return ans

File: C++/482. License Key Formatting.cpp
class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        int len = S.length(), cnt = 0;
        string ans = "";
        int first = 0;
        for(int i=len-1;i>=0;i--)
        {
            if(S[i]=='-')
                continue;
            if(cnt == K)
            {
                ans = '-' + ans;
                cnt = 0;
            }
            ans = (char)toupper(S[i]) + ans;
            cnt++;
        }
        return ans;
    }
};